org.uqbar.project.wollok.typeSystem/src/org/uqbar/project/wollok/typesystem/constraints/types/OffenderSelector.xtend

```diff
@@ -24,8 +24,8 @@ class OffenderSelector {
 
 	def static handleOffense(TypeVariable subtype, TypeVariable supertype, TypeSystemException offense) {
 		val offender = selectOffenderVariable(subtype, supertype)
-		offender.addError(offense)
-		offense.variable = offender
+		if (offense.variable === null) offense.variable = offender
```

Comment:
In which case is this null check needed?

Reply:
When we want select the offender from `selectOffenderVariable`, used for RejectedMinTypeException.
And.. for CannotBeVoidException, but in that case the variable can be set easily.